core/block.js

```diff
@@ -864,6 +881,11 @@ Blockly.Block.prototype.setColour = function(colour, colourSecondary, colourTert
     this.colourTertiary_ = goog.color.rgbArrayToHex(
         goog.color.darken(goog.color.hexToRgb(this.colour_), 0.2));
   }
+  if (colourQuaternary !== undefined) {
+    this.colourQuaternary_ = this.makeColour_(colourQuaternary);
+  } else {
+    this.colourQuaternary_ = this.colourTertiary_;
+  }
```

Comment:
What's the relationship between this `else` clause and the `throw` at the start of `colourHelper`?

Reply:
See my reply above about `colourHelper`. It would be called before this function as an extension to simplify defining the colors for block json.

`setColour` is used in many different places. I think I caught all of the places that were using 3 values and added the fourth. I do see at least one place (insertion marker in block.js) that was only using 1 value, so this conditional is used.